src/PrestaShopBundle/Controller/Api/Improve/Design/PositionsController.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Api\Improve\Design;
+
+use PrestaShopBundle\Controller\Api\ApiController;
+use PrestaShop\PrestaShop\Adapter\Validate;
+use PrestaShop\PrestaShop\Adapter\Module\Module;
+use PrestaShopBundle\Entity\Repository\CategoryRepository;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Module as LegacyModule;
+
+class PositionsController extends ApiController
+{
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function updateAction(Request $request)
+    {
+        $moduleId = (int) $request->request->get('moduleId');
```

Comment:
use $request->request->getInt('moduleId') instead

Reply:
you need to extract business logic from controller:

### easy strategy

* create a `ModulePositionUpdater` class with update($moduleId, $hookId, $way, $position) : array (empty if no errors)

### better strategy

* Create a new Object created from Request like `ModuleMovement($moduleId, $hookId, $way, $position)`. You can construct it from the request and invalidate it directly if $moduleId is invalid (an exemple here: https://github.com/PrestaShop/PrestaShop/pull/8984/files#diff-307533e1154f5cf404d67226aa79ef8eR43)
* Create `ModuleUpdater->move(ModuleMovement $moduleMovement)`, this function throw an exception if invalid with the already used error message.

In all cases, I discourage using directly classes from `Core` or `Adapter` namespaces in modern controllers and use of legacy `ObjectModel` is totally forbidden! I'll complete the docs.

Reply:
That's why I wouldn't create a PR from now. I know I have to do this, but just trying to understand how it currently works (still not found why there is some function / action in ajax). Please don't review for the moment =)